### DISCUSSION TITLE
navigation issue

### DISCUSSION THREAD

**Discussion by Eng1Mahmoud** [amannn/next-intl#1455] [Q&A]
i try use next-intl but i have some error >> 

![1](https://github.com/user-attachments/assets/182cdd90-78aa-408e-9ee2-28150f0af0fc)
![2](https://github.com/user-attachments/assets/fae3eeef-fb06-4592-be64-6c29813df6c7)
![3](https://github.com/user-attachments/assets/c77b35fa-ecb6-489a-933d-1c437d015ae1)
![code](https://github.com/user-attachments/assets/cc68c152-284e-4c1d-a8ec-c7ae31a9933a)

when I try navigate into /shop >> open >> http://localhost:5000/undefined/shop >> why this issue

**Comment by amannn** [comment]
Can you share a reproduction, ideally based on https://github.com/amannn/next-intl-bug-repro-app-router?

**Reply by Eng1Mahmoud** [reply]
can you explain more , what you mean , i follow next-intel docs  @amannn

**Reply by amannn** [reply]
I can't see a problem in the code you've provided, can you provide a github repository with the full code where the issue can be reproduced? (ideally using the template I've mentioned)

**Reply by Eng1Mahmoud** [reply]
the code exist in this branch >> https://github.com/Eng1Mahmoud/watch-store/tree/localization @amannn

**Reply by amannn** [reply]
Your i18n middleware doesn't execute where necessary. Note that it should execute for all content routes (e.g. `'/en'`).

There are two problems:
1. Your i18n middleware exports a [matcher](https://github.com/Eng1Mahmoud/watch-store/blob/b9f69fdd48ec740a77590e1e74804ae9ba122a78/middleware/LocalizationMiddleware.ts#L6-L9). This doesn't have any effect, your [main middleware](https://github.com/Eng1Mahmoud/watch-store/blob/localization/middleware.ts) should have this matcher.
2. Your auth middleware runs before the i18n middleware, always returning a response, therefore the i18n middleware never runs.

If you disable the auth middleware temporarily and you move the matcher, then the link works as expected.

I'll make sure to get better error handling in for this case, thank you for the report!

**Reply by Eng1Mahmoud** [reply]
oh, thanks, It works after I move the matcher into the main middleware @amannn

**Reply by Eng1Mahmoud** [reply]
i have anthore question , i want set translation for every page or component in separit file , 

like message/en/hom.json and etc how i can acheve this , i do it in i18next , but here i can not know if it work or not  @amannn

**Reply by amannn** [reply]
This is covered in the docs: [How can I split my messages into multiple files?](https://next-intl-docs.vercel.app/docs/usage/configuration#messages-split-files)

**Reply by Eng1Mahmoud** [reply]
i see it,  but i looks for dynamic way not need add every new file manual @amannn Thanks

**Reply by Eng1Mahmoud** [reply]
![2](https://github.com/user-attachments/assets/5d86da4a-1177-4fce-ac53-112016e0e1da)
![code](https://github.com/user-attachments/assets/32f6b974-3cf7-4737-b82f-e404d30dbb4e)
![code1](https://github.com/user-attachments/assets/c40426e4-1f4a-4933-8fb5-a414083d5a05)
only localizationMiddleware runs after finish break next middleware, how can I fix this  @amannn please help me
